xdsl/dialects/arm/assembly.py

```diff
@@ -2,11 +2,31 @@
 
 from xdsl.dialects.arm.register import ARMRegisterType
 from xdsl.dialects.builtin import StringAttr
-from xdsl.ir import SSAValue
+from xdsl.ir import (
+    Data,
+    SSAValue,
+)
+from xdsl.irdl import irdl_attr_definition
+from xdsl.parser import AttrParser
+from xdsl.printer import Printer
 
 AssemblyInstructionArg: TypeAlias = SSAValue
 
 
+@irdl_attr_definition
+class LabelAttr(Data[str]):
+    name = "arm.label"
+
+    @classmethod
+    def parse_parameter(cls, parser: AttrParser) -> str:
+        with parser.in_angle_brackets():
+            return parser.parse_str_literal()
+
+    def print_parameter(self, printer: Printer) -> None:
+        with printer.in_angle_brackets():
+            printer.print_string_literal(self.data)
+
+
```

Comment:
this should be in the dialect

Reply:
Maybe we need a new `.attrs` file with the attribute definitions

Reply:
I have moved it to a separate .attrs file. thank you